components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.ts

```diff
@@ -100,14 +110,33 @@ export class ResourceDropdownComponent implements OnInit, OnChanges {
 
   closeDropdown(): void {
     this.dropdownState = 'closed';
-    this.onDropdownClosing.emit(
-      this.resources.filter(r => r.checked).map(r => r.id));
+    const flattenedIDs = [].concat(
+      ...this.snapshotResources.map(
+        resource => resource.itemList.filter(r => r.checked).map(r => r.id)));
+    this.onDropdownClosing.emit(flattenedIDs);
```

Comment:
Returned data is still the same--a list of checked IDs. But the structure is more complicated so generating that list now takes a bit more work.